templates/terraform/constants/address.go.erb

```diff
@@ -0,0 +1,14 @@
+// All internal addresses default to a purpose of `GCE_ENDPOINT`, but it is not
+// allowed on external addresses, so we can't set the default on the field.
+func addressDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
We could implement this as a `CustomizeDiff` as well I think, setting the `new` value to the default when `address_type` is internal. It's little nicer than a DSF imo, because Terraform shows what it will do in plan, but isn't necessary. (this is _very_ much a stretch thing, and feel free to ignore it. Historically we've been a little squeamish about CD, so we don't have great examples of using it so I over-suggest it on PRs sometimes.)